src/tools/rbd_wnbd/wnbd_wmi.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (c) 2019 SUSE LLC
+ * Copyright (C) 2022 Cloudbase Solutions
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#pragma once
+#include <comutil.h>
+
+#define _WIN32_DCOM
+#include <wbemcli.h>
+
+#include <string>
+#include <vector>
+
+HRESULT co_initialize_basic();
+
+class WmiConnection
+{
+private:
+  std::wstring ns;
+public:
+  IWbemLocator* wbem_loc;
+  IWbemServices* wbem_svc;
+
+  WmiConnection()
+    : wbem_loc(nullptr)
+    , wbem_svc(nullptr)
+  {
+  }
```

Comment:
This constructor seems redundant to me as there is no way to set `ns` after default-constructing.

Reply:
"root/cimv2" would be the default namespace used by wmi, which can be useful.

Reply:
Do you mean that `IWbemLocator::ConnectServer` would default to `root\cimv2` if an empty `ns` is provided?  Is this documented somewhere?

Reply:
https://learn.microsoft.com/en-us/windows/win32/wmisdk/setting-namespace-security-with-the-wmi-control#setting-the-default-namespace-for-scripts

```
get-item hklm:\SOFTWARE\Microsoft\WBEM\Scripting

    Hive: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\WBEM

Name                           Property
----                           --------
Scripting                      Default Impersonation Level : 3
                               Default Namespace           : root\cimv2
```

However, since the default can be changed through a registry key (although that would probably break a lot of Windows apps :) ), it's safer to just define the default on our side.

Reply:
I'd just get rid of the defaulting constructor -- it's unused and it's much simpler that way ;)